Include/internal/pycore_pystate.h

```diff
@@ -138,13 +138,37 @@ _PyThreadState_GET(void)
 //
 // High-level code should generally call PyEval_RestoreThread() instead, which
 // calls this function.
-void _PyThreadState_Attach(PyThreadState *tstate);
+extern void _PyThreadState_Attach(PyThreadState *tstate);
 
 // Detaches the current thread from the interpreter.
 //
 // High-level code should generally call PyEval_SaveThread() instead, which
 // calls this function.
-void _PyThreadState_Detach(PyThreadState *tstate);
+extern void _PyThreadState_Detach(PyThreadState *tstate);
+
+// Temporarily pauses the thread in the GC state.
+//
+// This is used to implement stop-the-world pauses. The thread must be in the
+// "attached" state. It will switch to the "GC" state and pause until the
+// stop-the-world event completes, after which it will switch back to the
+// "attached" state.
+extern void _PyThreadState_Park(PyThreadState *tstate);
```

Comment:
"Park" is a relatively unfamiliar term.  Consider this alternative:

```suggestion
extern void _PyEval_PauseThread(PyThreadState *tstate);
```

Reply:
How about "suspend"? I think it has a bit more association with the idea that a different thread is in control.

I refactored the function to make it more clear that like `_PyThreadState_Detach()` and `_PyThreadState_Attach()` it corresponds to a state transition.